File: test/language/expressions/java/expressions/ConstTypeTest.java
package expressions;

import static org.junit.Assert.*;

import org.junit.Test;

import expressions.const_type.ConstTypeExpression;

public class ConstTypeTest
{
    @Test
    public void bitSizeOfWithOptional()
    {
        final ConstTypeExpression constTypeExpression = new ConstTypeExpression(VALID_VALUE, ADDITIONAL_VALUE);

        assertEquals(CONST_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL, constTypeExpression.bitSizeOf());
    }

    @Test
    public void bitSizeOfWithoutOptional()
    {
        final ConstTypeExpression constTypeExpression = new ConstTypeExpression();
        constTypeExpression.setValue(INVALID_VALUE);

        assertEquals(CONST_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL, constTypeExpression.bitSizeOf());
    }

    private static final int CONST_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL = 10;
    private static final int CONST_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL = 7;

    private static final byte VALID_VALUE = 0x01;
    private static final byte INVALID_VALUE = 0x00;

    private static final byte ADDITIONAL_VALUE = 0x03;
}

File: test/language/expressions/cpp98/ConstTypeTest.cpp
#include "math.h"

#include "gtest/gtest.h"

#include "expressions/const_type/ConstTypeExpression.h"

namespace expressions
{
namespace const_type
{

TEST(ConstTypeTest, bitSizeOfWithOptional)
{
    ConstTypeExpression constTypeExpression;
    const uint8_t validValue = 0x01;
    const uint8_t additionalValue = 0x03;
    constTypeExpression.setValue(validValue);
    constTypeExpression.setAdditionalValue(additionalValue);

    const size_t constTypeExpressionBitSizeWithOptional = 10;
    ASSERT_EQ(constTypeExpressionBitSizeWithOptional, constTypeExpression.bitSizeOf());
}

TEST(ConstTypeTest, bitSizeOfWithoutOptional)
{
    ConstTypeExpression constTypeExpression;
    const uint8_t invalidValue = 0x00;
    constTypeExpression.setValue(invalidValue);

    const size_t constTypeExpressionBitSizeWithoutOptional = 7;
    ASSERT_EQ(constTypeExpressionBitSizeWithoutOptional, constTypeExpression.bitSizeOf());
}

} // namespace const_type
} // namespace expressions
